crates/swc_ecma_transforms_typescript/src/typescript.rs

```diff
@@ -183,7 +183,12 @@ where
         }
 
         if !self.config.verbatim_module_syntax {
-            let span = n.span;
+            let span = match n.shebang.is_some() {
+                true => n
+                    .span
+                    .with_lo(n.body.first().map(|s| s.span_lo()).unwrap_or(n.span.lo)),
+                false => n.span,
+            };
```

Comment:
FWIW, I think probalby the jsx prag detection code should be consolidated, but I believe the jsx prag detection code in swc_ecma_transforms_react is doing too much work as it checks every module item instead of just the top of the file: https://github.com/swc-project/swc/blob/5fa2ed4bfdb49f77d4cb77004e1f61c8e2c36b64/crates/swc_ecma_transforms_react/src/jsx/mod.rs#L1033-L1040

Reply:
https://github.com/swc-project/swc/issues/2992#issuecomment-990739450

It's due to mdxjs